app/components/on-this-page.tsx

```diff
@@ -1,8 +1,26 @@
 import { Link } from "@remix-run/react";
 import { type Doc } from "~/modules/gh-docs/.server";
 import iconsHref from "~/icons.svg";
+import { useEffect, useState } from "react";
 
 export function LargeOnThisPage({ doc }: { doc: Doc }) {
+  const [activeHeading, setActiveHeading] = useState<string>("");
+  useEffect(() => {
+    const container = document.getElementsByClassName("md-prose");
+    const headings = container[0].querySelectorAll("h2");
+
+    function handleScroll() {
+      headings.forEach((h) => {
+        if (window.scrollY > h.offsetTop - 100) {
+          setActiveHeading(h.id);
+        }
+      });
+    }
+    window.addEventListener("scroll", handleScroll);
+    return () => {
+      window.removeEventListener("scroll", handleScroll);
+    };
+  });
```

Comment:
No reason to re-run this `useEffect` every time the state changes. Let me know if there's a bug this is addressing

```suggestion
  }, []);
```